File: backends/php-laravel/app/Domain/Invitation/ValueObjects/InvitationVo.php
<?php

/**
 * This is a program developed by BobTabo.
 *
 * Copyright (c) 2026 BobTabo. All Rights Reserved.
 */

declare(strict_types=1);

namespace App\Domain\Invitation\ValueObjects;

use App\Support\Traits\Getter;
use App\Support\ValueObjects\AbstractValueObject;

/**
 * 招待 URL・トークンをまとめたスナップショット ValueObject です。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 * @package App\Domain\Invitation\ValueObjects
 *
 * @method bool isFound() 招待が見つかったかどうか
 * @method ?string getUrl() 招待 URL（完全）
 * @method ?string getDisplayUrl() 招待 URL（省略表示用）
 * @method ?string getToken() 招待トークン
 */
class InvitationVo extends AbstractValueObject
{
    use Getter;

    private bool $found = false;
    private ?string $url = null;
    private ?string $displayUrl = null;
    private ?string $token = null;
}

File: backends/java-spring/src/main/java/com/authorization/domain/invitation/valueobjects/InvitationVo.java
/**
 * This is a program developed by BobTabo.
 *
 * Copyright (c) 2026 BobTabo. All Rights Reserved.
 */
package com.authorization.domain.invitation.valueobjects;

import com.authorization.support.valueobjects.AbstractValueObject;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 招待ValueObjectクラスです。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 */
@Data
@EqualsAndHashCode(callSuper = true)
public class InvitationVo extends AbstractValueObject {

    private boolean found;
    private String url;
    private String displayUrl;
    private String token;
}
